### ISSUE TITLE
chrome://inspect?tracing not working

### ISSUE THREAD

**Issue by primiano** [catapult-project/catapult#1884]
Recently the tracing UI seems to fall in a broken state on a weekly basis. This week it's the case for chrome://inspect/?tracing
Tried on ToT canary (49.0.2618.0).

Repro step:
- open chrome for android on the device
- open chrome://inspect?tracing on desktop
- click on record

What happens:
Nothing happens (not even a JS warning)

What I expect:
A dialog that prompts me which categories to enable etc etc.

**Comment by primiano** [comment]
@petrcermak @natduca @skyostil  FYI

I'm going to create a crbug and mark as beta blocker. We cannot ship android without chrome://inspect?tracing.

As a more general comment, should we make tracing part, at very least, of the set of manual QA tests that we do before releases? The branch point is soon, and we might have end up shipping an untraceable chrome.

**Comment by primiano** [comment]
Nevermind the "and we might have end up shipping an untraceable chrome."
See http://crbug.com/576292 it seems to be broken even in M47, so I guess we already shipped a broken one.

**Comment by petrcermak** [comment]
I'll give it a go :hammer:

**Comment by petrcermak** [comment]
The problem seems to be that neither resolver function inside <tt>InspectorTracingControllerClient.prototype.getCategories</tt> is ever called:

```
getCategories: function() {
  var res = this.conn_.req('Tracing.getCategories', {});  // This does get called.
  return res.then(function(result) {
    return result.categories;                             // This does NOT get called.
  }, function(err) {
    return [];                                            // This does NOT get called.
  });
},
```

I will continue my investigation in the morning (GMT).

**Comment by natduca** [comment]
Should we stand up a browser test for this?

**Comment by petrcermak** [comment]
I agree with @primiano that we should definitely have end-to-end tests for tracing in Chrome. *My impression* (given the finding described in my last comment) is that this is a **Chrome issue** (rather than a Trace Viewer one).

Investigation still pending...

**Comment by primiano** [comment]
browser test is going to be tricky: we have to mock the presence of a device over adb and then poke around with the UI. My fear is that it's going to be too fake (due to the mocking) or too flaky (due to UI changes). That's why I was suggesting to add to the QA manual test. At least would give us coverage on the releases.

**Comment by skyostil** [comment]
Do we know how devtools tests chrome://inspect? Perhaps we just extend those tests.

**Comment by natduca** [comment]
Maybe ask pfeldman?

**Comment by petrcermak** [comment]
I've just reassigned this to @anniesullie, who has recently been working on a chrome://tracing manual testing plan.
